hpx/runtime/serialization/detail/polymorphic_nonintrusive_factory.hpp

```diff
@@ -240,4 +263,31 @@ namespace hpx { namespace serialization { namespace detail
         return hpx_register_class_instance;                                   \
     }                                                                         \
 /**/
+#define HPX_SERIALIZATION_WITH_CUSTOM_CONSTRUCTOR(Class, Func)                \
+    namespace hpx { namespace serialization { namespace detail {              \
+    template<>                                                                \
+    class constructor_selector<HPX_UTIL_STRIP(Class)>                         \
+    {                                                                         \
+    public:                                                                   \
+        static Class *create(input_archive& ar)                               \
+        {                                                                     \
+            return Func(ar);                                                  \
+        }                                                                     \
+    };                                                                        \
+    }}}                                                                       \
+/**/
+#define HPX_SERIALIZATION_WITH_CUSTOM_CONSTRUCTOR_TEMPLATE(                   \
+    Parameters, Template, Func)                                               \
+    namespace hpx { namespace serialization { namespace detail {              \
+    HPX_UTIL_STRIP(Parameters)                                                \
+    class constructor_selector<HPX_UTIL_STRIP(Template)>                      \
+    {                                                                         \
+    public:                                                                   \
+        static HPX_UTIL_STRIP(Template) *create(input_archive& ar)            \
+        {                                                                     \
+            return Func(ar, static_cast<HPX_UTIL_STRIP(Template)*>(0));       \
```

Comment:
Again pasting off-line discussion: overloading is not the user case, but binding template parameters. Func() may be a templated function, or might not be -- so explicitly passing template parameters is not an option. A templated function has no way of knowing the values for its templates' parameters, so passing this dummy pointer is a means to deduce those.